12-rest/classworks/Egor_Emelyanov/index.js

```diff
@@ -0,0 +1,80 @@
+const Koa = require('koa');
+const router = require('koa-router');
+const bodyParser = require('koa-bodyparser');
+const joi = require('joi');
+const koaJwt = require('koa-jwt');
+const jwt = require('jsonwebtoken');
+const crypto = require('crypto');
+const fs = require('fs');
+const fetch = require('node-fetch');
+
+const secret = 'chikibamboni';
+
+const db = {};
+
+const app = new Koa();
+const publicRoutes = router();
+const privateRouter = router();
+const schema = joi.object({
+  login: joi.string().required(),
+  password: joi.string().required(),
+});
+const interviewSchema = joi.array().items(joi.string().pattern(/\?$/)).min(4);
+const articleSchema = joi.array().items(joi.string()).min(3);
+
+app.use(bodyParser());
+
+app.use(publicRoutes.routes());
+
+publicRoutes.post('/signup', async (ctx) => {
+  schema.validate(ctx.request.body);
+  db[ctx.request.body.login] = crypto.pbkdf2Sync(ctx.request.body.password, 'salt', 100000, 64, 'sha512').toString('hex');
```

Comment:
users should be also stored in file db, not in JS object db as it will be lost after server restart